homeassistant/components/switch/fritzdect.py

```diff
@@ -0,0 +1,151 @@
+
+    # Add all actors to hass
+    for actor in fritz.get_actors():
+        # Only add devices that support switching
+        if actor.has_switch:
+            data = FritzDectSwitchData(fritz, actor.actor_id)
+            add_devices([FritzDectSwitch(hass, data, actor.name)], True)
+
+
+class FritzDectSwitch(SwitchDevice):
+    """Representation of a FRITZ!DECT switch."""
+
+    def __init__(self, hass, data, name):
+        """Initialize the switch."""
+        self.units = hass.config.units
+        self.data = data
+        self._name = name
+
+    @property
+    def name(self):
+        """Return the name of the FRITZ!DECT switch, if any."""
+        return self._name
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        attrs = {}
+
+        if self.data.has_powermeter:
+            attrs[ATTR_TOTAL_CONSUMPTION] = "%.3f %s" % \
+              (float(self.data.total_consumption), ATTR_TOTAL_CONSUMPTION_UNIT)
+
+        if self.data.has_temperature:
+            attrs[ATTR_TEMPERATURE] = "%.1f %s" % \
+              (self.units.temperature(self.data.temperature, TEMP_CELSIUS),
+               self.units.temperature_unit)
+
+        return attrs
+
+    @property
+    def current_power_watt(self):
+        """Return the current power usage in Watt."""
+        try:
+            return float(self.data.current_consumption)
+        except ValueError:
+            return None
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        return self.data.state
+
+    def turn_on(self, **kwargs):
+        """Turn the switch on."""
+        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+        actor.switch_on()
+
+    def turn_off(self):
+        """Turn the switch off."""
+        actor = self.data.fritz.get_actor_by_ain(self.data.ain)
+        actor.switch_off()
+
+    def update(self):
+        """Get the latest data from the fritz box and updates the states."""
+        self.data.update()
+
+    @property
+    def state(self):
+        """Return the staterof the sensor."""
+        return self.data.current_consumption
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return ATTR_CURRENT_CONSUMPTION_UNIT
+
+
+class FritzDectSwitchData(object):
+    """Get the latest data from the fritz box."""
+
+    def __init__(self, fritz, ain):
+        """Initialize the data object."""
+        self.fritz = fritz
+        self.ain = ain
+        self.state = STATE_UNKNOWN
+        self.temperature = STATE_UNKNOWN
+        self.current_consumption = STATE_UNKNOWN
+        self.total_consumption = STATE_UNKNOWN
+        self.has_switch = STATE_UNKNOWN
+        self.has_temperature = STATE_UNKNOWN
+        self.has_powermeter = STATE_UNKNOWN
+
+    def update(self):
+        """Get the latest data from the fritz box."""
+        actor = self.fritz.get_actor_by_ain(self.ain)
```

Comment:
Catch the exception thrown by `requests`. `requests` belongs to Home Assistant's requirements, just import it.